contracts/ModuleRegistry.sol

```diff
@@ -3,198 +3,322 @@ pragma solidity ^0.4.24;
+        _;
+    }
+
+    /////////////////////////////
+    // Initialization
+    /////////////////////////////
+
+    // Constructor
+    constructor () public
     {
+
+    }
+
+    function initialize(address _polymathRegistry, address _owner) external payable {
+        require(!getBool(Encoder.getKey("initialised")));
+        require(_owner != address(0) && _polymathRegistry != address(0), "0x address is in-valid");
+        set(Encoder.getKey("polymathRegistry"), _polymathRegistry);
+        set(Encoder.getKey("owner"), _owner);
+        set(Encoder.getKey("initialised"), true);
     }
 
     /**
-     * @notice Called by a security token to check if the ModuleFactory is verified or appropriate custom module
-     * @dev ModuleFactory reputation increases by one every time it is deployed
+     * @notice Called by a SecurityToken to check if the ModuleFactory is verified or appropriate custom module
+     * @dev ModuleFactory reputation increases by one every time it is deployed(used) by a ST.
      * @dev Any module can be added during token creation without being registered if it is defined in the token proxy deployment contract
      * @dev The feature switch for custom modules is labelled "customModulesAllowed"
      * @param _moduleFactory is the address of the relevant module factory
      */
     function useModule(address _moduleFactory) external {
         // This if statement is required to be able to add modules from the token proxy contract during deployment
-        if (ISecurityTokenRegistry(securityTokenRegistry).isSecurityToken(msg.sender)) {
-            if (IFeatureRegistry(featureRegistry).getFeatureStatus("customModulesAllowed")) {
-                require(verified[_moduleFactory]||(Ownable(_moduleFactory).owner() == Ownable(msg.sender).owner()),
-                  "ModuleFactory must be verified or SecurityToken owner must be ModuleFactory owner");
+        if (ISecurityTokenRegistry(getAddress(Encoder.getKey('securityTokenRegistry'))).isSecurityToken(msg.sender)) {
+            if (IFeatureRegistry(getAddress(Encoder.getKey('featureRegistry'))).getFeatureStatus("customModulesAllowed")) {
+                require(getBool(Encoder.getKey('verified', _moduleFactory)) || IOwnable(_moduleFactory).owner() == IOwnable(msg.sender).owner(),
+                "ModuleFactory must be verified or SecurityToken owner must be ModuleFactory owner");
             } else {
-                require(verified[_moduleFactory], "ModuleFactory must be verified");
+                require(getBool(Encoder.getKey('verified', _moduleFactory)), "ModuleFactory must be verified");
             }
-            require(registry[_moduleFactory] != 0, "ModuleFactory type should not be 0");
-            reputation[_moduleFactory].push(msg.sender);
-            emit LogModuleUsed(_moduleFactory, msg.sender);
+            uint8[] memory _latestVersion = ISecurityTokenRegistry(getAddress(Encoder.getKey('securityTokenRegistry'))).getProtocolVersion();
```

Comment:
I think we should be getting the version of the ST, not the latest version from the STR - i.e.:
`uint8[] memory _latestVersion = ISecurityToken(msg.sender).getVersion();`